capstone/src/main/java/com/google/sps/servlets/GroupMemberServlet.java

```diff
@@ -26,68 +34,92 @@
   @Override
   // Adds a new member to a group 
   public void doPost(String userId, HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
+
+    String email = request.getParameter("email");
     Long groupId = Long.parseLong(request.getParameter("groupId"));
-    
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Entity group = this.getGroupFromId(response, groupId, datastore);
-    if (group != null && !userId.equals("")) {
+    Entity memberEntity = getMemberEntity(email, response, datastore);
+    if (memberEntity == null) return;
+    Entity group = ServletHelper.getEntityFromId(response, groupId, datastore, "Group");
+    if (group == null) return;
+
+    String memResponse = doPost_helper(response, datastore, groupId, memberEntity, group);
+  }
+
+  // If no account for email, return 'This email doesn't have an account'
+  // If user already in group, return 'User already in group'
+  // If user not it group, add user and return 'User added to group'
+  private String doPost_helper(HttpServletResponse response, DatastoreService datastore, Long groupId, Entity memberEntity, Entity group) {
```

Comment:
doPost_helper should be all camel case. Also, please rename the method since the name isn't very descriptive of what it is doing. possibly something like maybeAddUserToGroup